File: frontend/src/types/movie.ts
export interface Movie {
    id: number;
    title: string;
    image?: string;
    overview?: string;
    genres: string[];
    creatorId?: number;
    userEnglishLevel?: string;
}

export interface MovieFormData {
    title: string;
    image?: string;
    overview?: string;
    genres: string[];
}

export interface FormValues {
    title: string;
    image?: string;
    overview?: string;
    genres: string[];
}

export interface NewGenreData {
    name: string;
}

export interface MovieDetails {
    id: number;
    title: string;
    image?: string;
    overview?: string;
    genres: string[];
    creatorId?: number;
    userEnglishLevel?: string;
}

File: backend/src/main/java/co/backend/movie/Movie.java
package co.backend.movie;

import co.backend.genre.Genre;
import co.backend.user.User;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.util.ArrayList;
import java.util.List;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Data
@Table(name = "movie")
public class Movie {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;

    @Column(name = "creator_id")
    private Long creatorId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "creator_id", insertable = false, updatable = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    private User creator;

    @Lob
    @Basic(fetch = FetchType.LAZY)
    @Column(name = "image", columnDefinition = "LONGBLOB")
    private byte[] imageData;

    @Column(columnDefinition = "TEXT")
    private String overview;

    @Lob
    @Basic(fetch = FetchType.LAZY)
    @Column(columnDefinition = "LONGBLOB")
    private byte[] script;

    @ManyToMany
    @JoinTable(name = "movie_genre", joinColumns = @JoinColumn(name = "movie_id"), inverseJoinColumns = @JoinColumn(name = "genre_id"))
    private List<Genre> genres = new ArrayList<>();
}
